spectral_cube/lower_dimensional_structures.py

```diff
@@ -1018,6 +1027,21 @@ def with_fill_value(self, fill_value):
         """
         return self._new_spectrum_with(fill_value=fill_value)
 
+    def with_beam(self, beam):
```

Comment:
should this method be defined on some even lower-level object...?  maybe not, just raising the possibility